apps/api/src/modules/pilot/pilot.service.ts

```diff
@@ -309,7 +310,143 @@ export class PilotService {
    * @param user - The user to run the pilot for
    * @param sessionId - The ID of the session to run the pilot for
    */
-  async runPilot(user: User, sessionId: string, session?: PilotSession) {
+  async runPilot(
+    user: User,
+    sessionId: string,
+    session?: PilotSession,
+    mode?: 'subtask' | 'summary' | 'finalOutput',
```

Comment:
Consider wrapping it into `options` parameter with object type to facilitate adding new fields in the future.